Add initial docs for Serverless Span Autolinking (#26209)
DataDog/documentation: content/en/serverless/aws_lambda/distributed_tracing.md

Learn more about [tracing through .NET Azure serverless applications][15].

## Span Auto-linking
Datadog can automatically detect linked spans when certain parts of your request cannot propogate trace context, such as when a request triggers an [S3 Change Events][28], or [DynamoDB Streams][29]. These will appear in the Span Links tab in the Datadog Trace UI. 

> **Note:** Span auto-linking may not work if you are only ingesting a sample of your traces because the linked traces might be dropped before ingestion. To improve your chances of seeing auto-linked spans, increase your sample rate. 

Reviewer: There is also the matter of not having the upstream span pointers if the object or item was created without them.

> If you are viewing the request that originated before the change event and the linked trace is ingested, you will be able to see

Reviewer: what will we se? =]

Learn more about [tracing through .NET Azure serverless applications][15].

## Span Auto-linking

> Datadog automatically detects linked spans when segements of your asynchronous requests cannot propogate trace context, such as when a request triggers an [S3 Change Events][28], or [DynamoDB Streams][29]. These span links will appear in the [Span Links tab][30] in the Datadog Trace UI. 

Reviewer 2: ```suggestion
Datadog automatically detects linked spans when segments of your asynchronous requests cannot propagate trace context, such as when a request triggers an [S3 Change Events][28], or [DynamoDB Streams][29]. These span links appear in the [Span Links tab][30] of the Datadog Trace View. 
```

> **Note:** Span auto-linking may not link traces if you are only ingesting a sample of your traces because the linked traces might be dropped before ingestion. To improve your chances of seeing auto-linked spans, increase your sample rate. 

Reviewer 2: ```suggestion
**Note:** Span auto-linking may not link traces if you are only ingesting a sample of your traces, as the linked traces might be dropped before ingestion. To increase the likelihood of seeing auto-linked spans, consider increasing your sample rate.
```

> If you are viewing the request that originated before the change event and the linked trace is ingested, you can the linked span as a `Backward` link. 

Reviewer 2: ```suggestion
If you are viewing the request that originated before the change event and the linked trace is ingested, you can see the linked span as a `Backward` link. 
```

> If you are viewing the request that originated before the change event and the linked trace is ingested, you can see the linked span as a `Forward` link. 

Reviewer 2: Is this a link in the UI that says "Forward" or "Backward". Just want to make sure it's clear. Thanks!

Author: yeah it's a tag, it looks like this - 

<img width="915" alt="image" src="https://github.com/user-attachments/assets/fbff6c9a-00f5-41a9-9a2d-deb0ef5ea8b2">

This functionality is available for Python instrumented Lambda Functions on layer version 101 and above.

### DyanmoDB Change Stream Auto-linking

For [DyanmoDB Change Streams][29], Span Auto-linking supports the following operations:

- `PutItem`
- `UpdateItem`
- `DeleteItem`

> - `BatchWriteItem`

Reviewer: ```suggestion
- `BatchWriteItem`
- `TransactWriteItems`
```
